### ISSUE TITLE
Windows 11 - scene tab issue after render

### ISSUE THREAD

**Issue by rwbaer** [makehumancommunity/makehuman#206]
Nightly Build: 20220719 alpha (HEAD:f8ff3bc5)

Steps to Reproduce:
Start MH
Switch to Rendering | Scene and observe what it looks like for current lighting
Go toRendering | Render tab; click render
Return to Rendering | Scene and observe that the display is a shadow now

Is this related to #198 and/or #204?

![RenderIssue](https://user-images.githubusercontent.com/3155051/179863723-db93896b-ce88-4594-adaf-b6aee5dd7f15.png)

**Comment by kshepherd2013** [comment]
I have tried the same sequence on Ubuntu 22.04 and it works perfectly (no shadow at any point).  Also no errors in the log.
It sounds like an OpenGL problem in Qt on Windows - may be GPU specific and may be shader related.
We might get more information if you run with:
python3 makehuman.py --debugopengl --fullloggingopengl --debugnumpy
(if you haven't already tried this).
Also you might check your windows opengl drivers in case there is a more recent one.
I apologise if I am suggesting stuff you have already tried.  I am not sure what might cause this.

**Comment by rwbaer** [comment]
@kshepherd2013 thanks for the idea.
When started this way there is no error or warning in the logs.  My environment is:
python 3.9
PyOpenGL 3.1.6
PyQt5 5.15.7
numpy 1.23.1

There are a few lines on the console related to dict_keys and numpy that may make a difference (and I never remember seeing before), but I don't really know where  in the program these might be coming from.

There are some QWidget errors that probably shouldn't be happening.


Attribute Qt::AA_EnableHighDpiScaling must be set before QCoreApplication is created.
libpng warning: iCCP: known incorrect sRGB profile
QWidget::setMinimumSize: (/TextView) Negative sizes (0,-1) are not possible
QWidget::setMinimumSize: (/TextView) Negative sizes (0,-1) are not possible
QWidget::setMinimumSize: (/TextView) Negative sizes (0,-1) are not possible
qt.gui.icc: Unsupported ICC profile class 70727472
QPngHandler: Failed to parse ICC profile
qt.gui.icc: Unsupported ICC profile class 70727472
QPngHandler: Failed to parse ICC profile
qt.gui.icc: Unsupported ICC profile class 70727472
QPngHandler: Failed to parse ICC profile
qt.gui.icc: Unsupported ICC profile class 70727472
QPngHandler: Failed to parse ICC profile
qt.gui.icc: Unsupported ICC profile class 70727472
QPngHandler: Failed to parse ICC profile
qt.gui.icc: Unsupported ICC profile class 70727472
QPngHandler: Failed to parse ICC profile
qt.gui.icc: Unsupported ICC profile class 70727472
QPngHandler: Failed to parse ICC profile
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_long_Array_1'>])

**Comment by kshepherd2013** [comment]
Here is my complete log for running python3 makehuman.py --debugopengl --fullloggingopengl --debugnumpy
and then your exact sequence, and then exiting:
MakeHuman Copyright (C) 2001-2020 http://www.makehumancommunity.org
This program comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions. For details use the option --license

PRE STARTUP SETTING: useHDPI = False
PRE STARTUP SETTING: noShaders = False
PRE STARTUP SETTING: noSampleBuffers = False
Initialized logging
VERSION: render_tab_depth_fix:a4345b52
SHORT VERSION: alpha
LONG VERSION: 1.2.1 alpha (render_tab_depth_fix:a4345b52)
BASEMESH VERSION: hm08
IS BUILT (FROZEN): No
IS RELEASE VERSION: No
DEFAULT ENCODING: utf-8
FILESYSTEM ENCODING: utf-8
STDOUT ENCODING: utf-8
LOCALE PREFERRED ENCODING: UTF-8
WORKING DIRECTORY: /home/kshepherd/makehuman-devel/makehuman/makehuman
HOME LOCATION: /home/kshepherd/
PYTHON PATH: ./:./lib:./apps:./shared:./apps/gui:./core:/home/kshepherd/makehuman-devel/makehuman/makehuman:/usr/lib/python310.zip:/usr/lib/python3.10:/usr/lib/python3.10/lib-dynload:/home/kshepherd/.local/lib/python3.10/site-packages:/usr/local/lib/python3.10/dist-packages:/usr/lib/python3/dist-packages:/usr/lib/petsc/lib/python3/dist-packages:/usr/lib/slepc/lib/python3/dist-packages:/usr/lib/python3.10/dist-packages
DLL PATH: /home/kshepherd/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/home/kshepherd/scarletline/cyber/src/scripts/sh:/home/kshepherd/scarletline/cyber/src/scripts/perl:/home/kshepherd/scarletline/cyber/dev/Android/Sdk/ndk-bundle:/home/kshepherd/scarletline/cyber/dev/Android/Sdk/platform-tools:/home/kshepherd/scarletline/cyber/dev/Android/Sdk/tools
SYS.VERSION: 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]
SYS.PLATFORM: linux
SYS.EXECUTABLE: /usr/bin/python3
PLATFORM.MACHINE: x86_64
PLATFORM.PROCESSOR: x86_64
PLATFORM.UNAME.RELEASE: 5.15.0-35-generic
PLATFORM.LINUX_DISTRIBUTION: Ubuntu 22.04 Jammy Jellyfish
NUMPY.VERSION: 1.21.5
Loaded libGL.so => libGL.so.1 <CDLL 'libGL.so.1', handle 55a2e52cf050 at 0x7ffb4052a320>
No OpenGL_accelerate module loaded: No module named 'OpenGL_accelerate'
Loaded libGLU.so => libGLU.so.1 <CDLL 'libGLU.so.1', handle 55a2e530cfe0 at 0x7ffb405bd870>
QT.VERSION: 5.15.3
QT.JPG_SUPPORT: supported
QT.SVG_SUPPORT: supported
QT.PLUGIN_PATH: /usr/lib/x86_64-linux-gnu/qt5/plugins:/usr/bin
QT.PLUGIN_PATH_ENV: 
QT.CONF: NOT PRESENT
PYOPENGL.VERSION: 3.1.5
NOSHADERS: not set
NOSAMPLEBUFFERS: not set
GL.VENDOR: Intel
GL.RENDERER: Mesa Intel(R) UHD Graphics 630 (CFL GT2)
GL.VERSION: 4.6 (Compatibility Profile) Mesa 22.0.1
4.60
GLSL.VERSION: 4.60
GL.EXTENSION: GL_ARB_multisample not available (0x samples)
Creating new texture for image data/textures/texture_notfound.png.
GL.EXTENSION: GL_ARB_texture_non_power_of_two enabled
initTexture: 256, 256, False
Setting language to english
Setting language to english
Initializing default theme first.
Theme default version 1.2.0
Loaded theme data/themes/default.mht
Using Qt system style breeze
Loading human
compiled file missing: data/3dobjs/base.npz
Problem loading binary mesh: ('compiled file missing: %s', 'data/3dobjs/base.npz')
Loading ASCII mesh data/3dobjs/base.obj.
Calculated maximum number of faces for one vertex: 5
Calculated maximum number of poles for one vertex: 5
Not writing compiled meshes to system paths (data/3dobjs/base.npz).
Loading material from file data/skins/default.mhmat
Shader: adding built-in uniform b'gl_NormalMatrix'
Shader: adding built-in uniform b'gl_ModelViewProjectionMatrix'
VectorUniform(AdditiveShading) = [0.]
VectorUniform(AdditiveShading) = [0.]
Loaded vertex weights MakeHuman weights from file /home/kshepherd/makehuman-devel/makehuman/makehuman/data/rigs/default_weights.mhw
Loading scene
Loading scene file: data/scenes/default.mhscene
Loading main GUI
Assigning setting loaddir value /nfs/shepherd4_Primedia/Games/makehuman_home/makehuman/v1py3/models that was loaded before the setting was declared.
Assigning setting savedir value /nfs/shepherd4_Primedia/Games/makehuman_home/makehuman/v1py3/models/Caldonia_Watson that was loaded before the setting was declared.
Assigning setting exportdir value /nfs/shepherd4_Primedia/Games/makehuman_home/makehuman/v1py3/exports that was loaded before the setting was declared.
Loading plugins
Importing plugin 0_modeling_0_modifiers
Imported plugin 0_modeling_0_modifiers
Loading plugin 0_modeling_0_modifiers
Loading modifiers from data/modifiers/modeling_modifiers.json
Attempting to load targets from NPZ file.
Could not load targets from NPZ, loading individual files from data (Error message: ('Could not load load targets from npz archive. Archive file %s not found.', 'data/targets.npz'))
1280 targets loaded from .target files.
Loaded 249 modifiers from file data/modifiers/modeling_modifiers.json
Loaded 249 modifier descriptions from file data/modifiers/modeling_modifiers_desc.json
Loaded plugin 0_modeling_0_modifiers
Importing plugin 0_modeling_1_bodyshapes
Imported plugin 0_modeling_1_bodyshapes
Loading plugin 0_modeling_1_bodyshapes
Loading modifiers from data/modifiers/bodyshapes_modifiers.json
Loaded 22 modifiers from file data/modifiers/bodyshapes_modifiers.json
Loaded 22 modifier descriptions from file data/modifiers/bodyshapes_modifiers_desc.json
Loaded plugin 0_modeling_1_bodyshapes
Importing plugin 0_modeling_8_random
Imported plugin 0_modeling_8_random
Loading plugin 0_modeling_8_random
Loaded plugin 0_modeling_8_random
Importing plugin 0_modeling_9_custom_targets
Imported plugin 0_modeling_9_custom_targets
Loading plugin 0_modeling_9_custom_targets
Loaded plugin 0_modeling_9_custom_targets
Importing plugin 0_modeling_a_measurement
Imported plugin 0_modeling_a_measurement
Loading plugin 0_modeling_a_measurement
Loading modifiers from data/modifiers/measurement_modifiers.json
Loaded 20 modifiers from file data/modifiers/measurement_modifiers.json
Loaded 20 modifier descriptions from file data/modifiers/measurement_modifiers_desc.json
Loaded plugin 0_modeling_a_measurement
Importing plugin 0_modeling_background
Imported plugin 0_modeling_background
Loading plugin 0_modeling_background
Loaded plugin 0_modeling_background
Importing plugin 1_convert_model_tags
Imported plugin 1_convert_model_tags
Loading plugin 1_convert_model_tags
Loaded plugin 1_convert_model_tags
Importing plugin 1_mhapi
Imported plugin 1_mhapi
Loading plugin 1_mhapi
Loaded plugin 1_mhapi
Importing plugin 2_foot_posing
Imported plugin 2_foot_posing
Loading plugin 2_foot_posing
Loaded plugin 2_foot_posing
Importing plugin 2_posing_expression
Imported plugin 2_posing_expression
Loading plugin 2_posing_expression
Loaded plugin 2_posing_expression
Importing plugin 3_libraries_animation
Imported plugin 3_libraries_animation
Loading plugin 3_libraries_animation
Loaded plugin 3_libraries_animation
Importing plugin 3_libraries_clothes_chooser
Imported plugin 3_libraries_clothes_chooser
Loading plugin 3_libraries_clothes_chooser
Loaded plugin 3_libraries_clothes_chooser
Importing plugin 3_libraries_eye_chooser
Imported plugin 3_libraries_eye_chooser
Loading plugin 3_libraries_eye_chooser
According to mh.getSysDataPath here's high poly eyes data/eyes/high-poly/high-poly.mhclo
Selecting proxy file "data/eyes/high-poly/high-poly.mhclo" from eyes library.
compiled proxy file missing: data/eyes/high-poly/high-poly.mhpxy
Problem loading binary proxy: ('compiled proxy file missing: %s', 'data/eyes/high-poly/high-poly.mhpxy')
Loading proxy file: data/eyes/high-poly/high-poly.mhclo.
Loading material from file /home/kshepherd/makehuman-devel/makehuman/makehuman/data/eyes/materials/brown.mhmat
Shader: adding built-in uniform b'gl_NormalMatrix'
Shader: adding built-in uniform b'gl_ModelViewProjectionMatrix'
VectorUniform(AdditiveShading) = [0.]
VectorUniform(AdditiveShading) = [0.]
Not writing compiled proxies to system paths (data/eyes/high-poly/high-poly.mhpxy).
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/eyes/high-poly/high-poly.npz
Problem loading binary mesh: ('compiled file missing: %s', '/home/kshepherd/makehuman-devel/makehuman/makehuman/data/eyes/high-poly/high-poly.npz')
Loading ASCII mesh /home/kshepherd/makehuman-devel/makehuman/makehuman/data/eyes/high-poly/high-poly.obj.
Calculated maximum number of faces for one vertex: 4
Calculated maximum number of poles for one vertex: 4
Not writing compiled meshes to system paths (/home/kshepherd/makehuman-devel/makehuman/makehuman/data/eyes/high-poly/high-poly.npz).
remapping weights for proxy HighPolyEyes
remapping weights for HighPolyEyes took 0.08053 seconds
Loaded plugin 3_libraries_eye_chooser
Importing plugin 3_libraries_eyebrows
Imported plugin 3_libraries_eyebrows
Loading plugin 3_libraries_eyebrows
Loaded plugin 3_libraries_eyebrows
Importing plugin 3_libraries_eyelashes
Imported plugin 3_libraries_eyelashes
Loading plugin 3_libraries_eyelashes
Loaded plugin 3_libraries_eyelashes
Importing plugin 3_libraries_material_chooser
Imported plugin 3_libraries_material_chooser
Loading plugin 3_libraries_material_chooser
Loaded plugin 3_libraries_material_chooser
Importing plugin 3_libraries_polygon_hair_chooser
Imported plugin 3_libraries_polygon_hair_chooser
Loading plugin 3_libraries_polygon_hair_chooser
Loaded plugin 3_libraries_polygon_hair_chooser
Importing plugin 3_libraries_pose
Imported plugin 3_libraries_pose
Loading plugin 3_libraries_pose
Loaded plugin 3_libraries_pose
Importing plugin 3_libraries_proxy_chooser
Imported plugin 3_libraries_proxy_chooser
Loading plugin 3_libraries_proxy_chooser
Loaded plugin 3_libraries_proxy_chooser
Importing plugin 3_libraries_skeleton
Imported plugin 3_libraries_skeleton
Loading plugin 3_libraries_skeleton
Loaded plugin 3_libraries_skeleton
Importing plugin 3_libraries_teeth
Imported plugin 3_libraries_teeth
Loading plugin 3_libraries_teeth
Loaded plugin 3_libraries_teeth
Importing plugin 3_libraries_tongue
Imported plugin 3_libraries_tongue
Loading plugin 3_libraries_tongue
Loaded plugin 3_libraries_tongue
Importing plugin 4_rendering_9_viewer
Imported plugin 4_rendering_9_viewer
Loading plugin 4_rendering_9_viewer
Loaded plugin 4_rendering_9_viewer
Importing plugin 4_rendering_opengl
Imported plugin 4_rendering_opengl
Loading plugin 4_rendering_opengl
Assigning setting rendering_width value 800 that was loaded before the setting was declared.
Assigning setting rendering_height value 600 that was loaded before the setting was declared.
Assigning setting GL_RENDERER_SSS value False that was loaded before the setting was declared.
Assigning setting GL_RENDERER_AA value True that was loaded before the setting was declared.
Loaded plugin 4_rendering_opengl
Importing plugin 4_rendering_scene
Imported plugin 4_rendering_scene
Loading plugin 4_rendering_scene
Setting rendering_width is already declared. Adding it again has no effect.
Setting rendering_height is already declared. Adding it again has no effect.
Loaded plugin 4_rendering_scene
Importing plugin 5_settings_0_settings
Imported plugin 5_settings_0_settings
Loading plugin 5_settings_0_settings
Setting language to english
Theme makehuman version 1.2.0
Loaded theme data/themes/makehuman.mht
Loaded Qt style data/themes/makehuman.qss
Loaded plugin 5_settings_0_settings
Importing plugin 5_settings_mouse
Imported plugin 5_settings_mouse
Loading plugin 5_settings_mouse
Loaded plugin 5_settings_mouse
Importing plugin 5_settings_shortcuts
Imported plugin 5_settings_shortcuts
Loading plugin 5_settings_shortcuts
Loaded plugin 5_settings_shortcuts
Importing plugin 5_settings_userplugins
Imported plugin 5_settings_userplugins
Loading plugin 5_settings_userplugins
Loaded plugin 5_settings_userplugins
Importing plugin 6_help
Imported plugin 6_help
Loading plugin 6_help
Loaded plugin 6_help
Importing plugin 7_data
Imported plugin 7_data
Loading plugin 7_data
Loaded plugin 7_data
Importing plugin 7_example
Imported plugin 7_example
Loading plugin 7_example
Loaded plugin 7_example
Importing plugin 7_expression_mixer
Imported plugin 7_expression_mixer
Loading plugin 7_expression_mixer
Loaded plugin 7_expression_mixer
Importing plugin 7_logging
Imported plugin 7_logging
Loading plugin 7_logging
Loaded plugin 7_logging
Importing plugin 7_material_editor
Imported plugin 7_material_editor
Loading plugin 7_material_editor
Loaded plugin 7_material_editor
Importing plugin 7_profile
Imported plugin 7_profile
Loading plugin 7_profile
Loaded plugin 7_profile
Importing plugin 7_save_targets
Imported plugin 7_save_targets
Loading plugin 7_save_targets
Loaded plugin 7_save_targets
Importing plugin 7_scene_editor
Imported plugin 7_scene_editor
Loading plugin 7_scene_editor
Setting rendering_width is already declared. Adding it again has no effect.
Setting rendering_height is already declared. Adding it again has no effect.
Assigning setting Scene_Editor_FileDlgPath value /nfs/shepherd4_Primedia/Games/makehuman_home/makehuman/v1py3/data/scenes that was loaded before the setting was declared.
Loaded plugin 7_scene_editor
Importing plugin 7_scripting
Imported plugin 7_scripting
Loading plugin 7_scripting
Loaded plugin 7_scripting
Importing plugin 7_shell
Imported plugin 7_shell
Loading plugin 7_shell
Using selector: EpollSelector
Using selector: EpollSelector

*** MESSAGE TYPE:kernel_info_request***
   Content: {}
   --->
   
kernel_info_request: {'header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_0', 'msg_type': 'kernel_info_request', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 385373, tzinfo=tzutc()), 'version': '5.3'}, 'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_0', 'msg_type': 'kernel_info_request', 'parent_header': {}, 'metadata': {}, 'content': {}, 'buffers': []}
{'header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_1', 'msg_type': 'kernel_info_reply', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 391781, tzinfo=datetime.timezone.utc), 'version': '5.3'}, 'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_1', 'msg_type': 'kernel_info_reply', 'parent_header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_0', 'msg_type': 'kernel_info_request', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 385373, tzinfo=tzutc()), 'version': '5.3'}, 'content': {'status': 'ok', 'protocol_version': '5.3', 'implementation': 'ipython', 'implementation_version': '7.31.1', 'language_info': {'name': 'python', 'version': '3.10.4', 'mimetype': 'text/x-python', 'codemirror_mode': {'name': 'ipython', 'version': 3}, 'pygments_lexer': 'ipython3', 'nbconvert_exporter': 'python', 'file_extension': '.py'}, 'banner': "Python 3.10.4 (main, Jun 29 2022, 12:14:53) [GCC 11.2.0]\nType 'copyright', 'credits' or 'license' for more information\nIPython 7.31.1 -- An enhanced Interactive Python. Type '?' for help.\n", 'help_links': [{'text': 'Python Reference', 'url': 'https://docs.python.org/3.10'}, {'text': 'IPython Reference', 'url': 'https://ipython.org/documentation.html'}, {'text': 'NumPy Reference', 'url': 'https://docs.scipy.org/doc/numpy/reference/'}, {'text': 'SciPy Reference', 'url': 'https://docs.scipy.org/doc/scipy/reference/'}, {'text': 'Matplotlib Reference', 'url': 'https://matplotlib.org/contents.html'}, {'text': 'SymPy Reference', 'url': 'http://docs.sympy.org/latest/index.html'}, {'text': 'pandas Reference', 'url': 'https://pandas.pydata.org/pandas-docs/stable/'}]}, 'metadata': {}, 'tracker': <zmq.sugar.tracker.MessageTracker object at 0x7ffaedccc520>}

*** MESSAGE TYPE:history_request***
   Content: {'raw': True, 'output': False, 'hist_access_type': 'tail', 'n': 1000}
   --->
   
history_request: {'header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_1', 'msg_type': 'history_request', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 395929, tzinfo=tzutc()), 'version': '5.3'}, 'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_1', 'msg_type': 'history_request', 'parent_header': {}, 'metadata': {}, 'content': {'raw': True, 'output': False, 'hist_access_type': 'tail', 'n': 1000}, 'buffers': []}
{'header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_4', 'msg_type': 'history_reply', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 426895, tzinfo=datetime.timezone.utc), 'version': '5.3'}, 'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_4', 'msg_type': 'history_reply', 'parent_header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_1', 'msg_type': 'history_request', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 395929, tzinfo=tzutc()), 'version': '5.3'}, 'content': {'status': 'ok', 'history': []}, 'metadata': {}, 'tracker': <zmq.sugar.tracker.MessageTracker object at 0x7ffaedccc520>}
Loaded plugin 7_shell

*** MESSAGE TYPE:execute_request***
   Content: {'code': '', 'silent': True, 'store_history': True, 'user_expressions': {}, 'allow_stdin': True}
   --->
   
execute_request: {'header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_2', 'msg_type': 'execute_request', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 493593, tzinfo=tzutc()), 'version': '5.3'}, 'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_2', 'msg_type': 'execute_request', 'parent_header': {}, 'metadata': {}, 'content': {'code': '', 'silent': True, 'store_history': True, 'user_expressions': {}, 'allow_stdin': True}, 'buffers': []}
{'header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_7', 'msg_type': 'execute_reply', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 508978, tzinfo=datetime.timezone.utc), 'version': '5.3'}, 'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_7', 'msg_type': 'execute_reply', 'parent_header': {'msg_id': 'be7339ad-704fb091b1120d80ae0bf6e7_1715437_2', 'msg_type': 'execute_request', 'username': 'kshepherd', 'session': 'be7339ad-704fb091b1120d80ae0bf6e7', 'date': datetime.datetime(2022, 7, 25, 15, 9, 53, 493593, tzinfo=tzutc()), 'version': '5.3'}, 'content': {'status': 'ok', 'execution_count': 0, 'user_expressions': {}, 'payload': []}, 'metadata': {'started': datetime.datetime(2022, 7, 25, 15, 9, 53, 498058, tzinfo=datetime.timezone.utc), 'dependencies_met': True, 'engine': '5aadf972-fa04-4499-9b5f-9aad2074eb2a', 'status': 'ok'}, 'tracker': <zmq.sugar.tracker.MessageTracker object at 0x7ffaedccc520>}
Received history reply with 0 entries
Importing plugin 7_targets
Imported plugin 7_targets
Loading plugin 7_targets
Loaded plugin 7_targets
Importing plugin 8_asset_downloader
Imported plugin 8_asset_downloader
Loading plugin 8_asset_downloader
Loaded plugin 8_asset_downloader
Importing plugin 8_server_socket
Imported plugin 8_server_socket
Loading plugin 8_server_socket
Loaded plugin 8_server_socket
Importing plugin 9_export_bvh
Imported plugin 9_export_bvh
Loading plugin 9_export_bvh
Loaded plugin 9_export_bvh
Importing plugin 9_export_collada
Imported plugin 9_export_collada
Loading plugin 9_export_collada
Loaded plugin 9_export_collada
Importing plugin 9_export_fbx
Imported plugin 9_export_fbx
Loading plugin 9_export_fbx
Loaded plugin 9_export_fbx
Importing plugin 9_export_light
Imported plugin 9_export_light
Loading plugin 9_export_light
Loaded plugin 9_export_light
Importing plugin 9_export_obj
Imported plugin 9_export_obj
Loading plugin 9_export_obj
Loaded plugin 9_export_obj
Importing plugin 9_export_ogre
Imported plugin 9_export_ogre
Loading plugin 9_export_ogre
Loaded plugin 9_export_ogre
Importing plugin 9_export_stl
Imported plugin 9_export_stl
Loading plugin 9_export_stl
Loaded plugin 9_export_stl
Importing plugin 9_export_uvmap
Imported plugin 9_export_uvmap
Loading plugin 9_export_uvmap
Loaded plugin 9_export_uvmap
Importing plugin 9_massproduce
Imported plugin 9_massproduce
Loading plugin 9_massproduce
Loaded plugin 9_massproduce
Loading GUI
showing category Modelling
showing task Macro modelling
Loading theme
Theme makehuman version 1.2.0
Loaded theme data/themes/makehuman.mht
Loaded Qt style data/themes/makehuman.qss
Applying targets
no compressed targets found
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/targets/macrodetails/african-female-young.target
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/targets/macrodetails/african-male-young.target
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/targets/macrodetails/asian-female-young.target
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/targets/macrodetails/asian-male-young.target
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/targets/macrodetails/caucasian-female-young.target
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/targets/macrodetails/caucasian-male-young.target
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/targets/macrodetails/universal-female-young-averagemuscle-averageweight.target
compiled file missing: /home/kshepherd/makehuman-devel/makehuman/makehuman/data/targets/macrodetails/universal-male-young-averagemuscle-averageweight.target
Updating skeleton joint positions
Done baking animations
Loading macro targets
Loading done

libpng warning: iCCP: known incorrect sRGB profile
Rendering mesh rectangle_None
Rendering mesh base.obj
Creating new texture for dynamic image data/litspheres/adaptive_skin_tone.png.
initTexture: 256, 256, False
Rendering mesh back_grid
Rendering mesh high-poly.obj
Creating new texture for image /home/kshepherd/makehuman-devel/makehuman/makehuman/data/eyes/materials/brown_eye.png.
initTexture: 1024, 1024, False
Creating new texture for image /home/kshepherd/makehuman-devel/makehuman/makehuman/data/litspheres/skinmat_eye.png.
initTexture: 256, 256, False
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
hiding category Modelling
showing category Rendering
hiding task Macro modelling
showing task Render
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
hiding task Render
showing task Scene
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
hiding task Scene
showing task Render
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering at 800x600
Rendering mesh base.obj
Rendering mesh high-poly.obj
Rendering mesh base.obj
Rendering mesh high-poly.obj
Image "<image.Image object at 0x7ffae17fb0a0>" loaded in image viewer.
hiding task Render
showing task Viewer
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
hiding task Viewer
showing task Scene
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
Rendering mesh rectangle_None
Rendering mesh base.obj
Rendering mesh back_grid
Rendering mesh high-poly.obj
ABOUT TO STOP APPLICATION
Unloading plugin 0_modeling_0_modifiers
Unloaded plugin 0_modeling_0_modifiers
Unloading plugin 0_modeling_1_bodyshapes
Unloaded plugin 0_modeling_1_bodyshapes
Unloading plugin 0_modeling_8_random
Unloaded plugin 0_modeling_8_random
Unloading plugin 0_modeling_9_custom_targets
Unloaded plugin 0_modeling_9_custom_targets
Unloading plugin 0_modeling_a_measurement
Unloaded plugin 0_modeling_a_measurement
Unloading plugin 0_modeling_background
Unloaded plugin 0_modeling_background
Unloading plugin 1_convert_model_tags
Unloaded plugin 1_convert_model_tags
Unloading plugin 1_mhapi
Unloaded plugin 1_mhapi
Unloading plugin 2_foot_posing
Unloaded plugin 2_foot_posing
Unloading plugin 2_posing_expression
Unloaded plugin 2_posing_expression
Unloading plugin 3_libraries_animation
Unloaded plugin 3_libraries_animation
Unloading plugin 3_libraries_clothes_chooser
Unloaded plugin 3_libraries_clothes_chooser
Unloading plugin 3_libraries_eye_chooser
Unloaded plugin 3_libraries_eye_chooser
Unloading plugin 3_libraries_eyebrows
Unloaded plugin 3_libraries_eyebrows
Unloading plugin 3_libraries_eyelashes
Unloaded plugin 3_libraries_eyelashes
Unloading plugin 3_libraries_material_chooser
Unloaded plugin 3_libraries_material_chooser
Unloading plugin 3_libraries_polygon_hair_chooser
Unloaded plugin 3_libraries_polygon_hair_chooser
Unloading plugin 3_libraries_pose
Unloaded plugin 3_libraries_pose
Unloading plugin 3_libraries_proxy_chooser
Unloaded plugin 3_libraries_proxy_chooser
Unloading plugin 3_libraries_skeleton
Unloaded plugin 3_libraries_skeleton
Unloading plugin 3_libraries_teeth
Unloaded plugin 3_libraries_teeth
Unloading plugin 3_libraries_tongue
Unloaded plugin 3_libraries_tongue
Unloading plugin 4_rendering_9_viewer
Unloaded plugin 4_rendering_9_viewer
Unloading plugin 4_rendering_opengl
Unloaded plugin 4_rendering_opengl
Unloading plugin 4_rendering_scene
Unloaded plugin 4_rendering_scene
Unloading plugin 5_settings_0_settings
Unloaded plugin 5_settings_0_settings
Unloading plugin 5_settings_mouse
Unloaded plugin 5_settings_mouse
Unloading plugin 5_settings_shortcuts
Unloaded plugin 5_settings_shortcuts
Unloading plugin 5_settings_userplugins
Unloaded plugin 5_settings_userplugins
Unloading plugin 6_help
Unloaded plugin 6_help
Unloading plugin 7_data
Unloaded plugin 7_data
Unloading plugin 7_example
Unloaded plugin 7_example
Unloading plugin 7_expression_mixer
Unloaded plugin 7_expression_mixer
Unloading plugin 7_logging
Unloaded plugin 7_logging
Unloading plugin 7_material_editor
Unloaded plugin 7_material_editor
Unloading plugin 7_profile
Unloaded plugin 7_profile
Unloading plugin 7_save_targets
Unloaded plugin 7_save_targets
Unloading plugin 7_scene_editor
Unloaded plugin 7_scene_editor
Unloading plugin 7_scripting
Unloaded plugin 7_scripting
Unloading plugin 7_shell
Unloaded plugin 7_shell
Unloading plugin 7_targets
Unloaded plugin 7_targets
Unloading plugin 8_asset_downloader
Unloaded plugin 8_asset_downloader
Unloading plugin 8_server_socket
Unloaded plugin 8_server_socket
Unloading plugin 9_export_bvh
Unloaded plugin 9_export_bvh
Unloading plugin 9_export_collada
Unloaded plugin 9_export_collada
Unloading plugin 9_export_fbx
Unloaded plugin 9_export_fbx
Unloading plugin 9_export_light
Unloaded plugin 9_export_light
Unloading plugin 9_export_obj
Unloaded plugin 9_export_obj
Unloading plugin 9_export_ogre
Unloaded plugin 9_export_ogre
Unloading plugin 9_export_stl
Unloaded plugin 9_export_stl
Unloading plugin 9_export_uvmap
Unloaded plugin 9_export_uvmap
Unloading plugin 9_massproduce
Unloaded plugin 9_massproduce
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_int_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_int_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_int_Array_1'>])
dict_keys([<class 'numpy.ndarray'>, <class 'ctypes.c_void_p'>, <class 'bytes'>, <class 'OpenGL.converters.c_int_Array_1'>])

**Comment by rwbaer** [comment]
Build f8ff31c5
Okay. I've been able to confirm this "shadow" behavior on at least 2 work computers and one home computer with various NVidia or Intel 630 graphics cards under Windows. numpy doesn't see critical having tried at least 1.23.1 and 1.18.1.  pyopengl doesn't seem critical as it happen on both 3.1.0 and 3.1.6.  PyQt5 version does not seem critical as it occurs with both 5.12.7 and 5.15.7.  Python version does not seem critical because it happens with 3.7.6, 3.9 and 3.10.  We are left with it is something unique to how either PYQt5 or openGL runs under Windows rather than Linux.

It would be nice if one of the other devs could confirm that they are seeing the differing behaviors Kevin is experiencing on Linux and I am experiencein on Windows 11.  Also, does it  happen on Windows 10 as well?  It would also be nice to know  what happens if we rollback to the version before we started to fix the 22.04 version to know if it is related to the fixes or is just an previously undiscovered behavior because rendering in MH is relatively uncommon.  I could use a git refresher on exactly what would be the proper way to do that if one of you git experts can help.

**Comment by rwbaer** [comment]
I found a machine with Windows MH1.2.0 (20210426 final (HEAD:409e74d4) in which rendering is still working as expected.

**Comment by rwbaer** [comment]
As of build 1.2.1 alpha (master:1c4fa878) there is no shadow problem on Windows after render

**Comment by rwbaer** [comment]
1.2.1 alpha (master:d7e0c21a) 7/14/2022 2:04  AM
I have now be able to verity that the shadow problem first appears with"Merge pull request #200 from ksheperd2013"
Prior to that, the render tab does not produce a visible render, but at least the other tabs show a properly redered human.  I think this commit should be reverted if a fix is not obvious.  We can live with a "failing render tab" but not a completely broken interface on Windows

**Comment by rwbaer** [comment]
While it does seem we need to move to QTopenGLWidget, we may need to study the subtle changes that have happened to buffer updating and clipping to  make sure we accomodate our complex GUI.
See https://doc.qt.io/qt-5/qopenglwidget.html  and particularly:
> https://doc.qt.io/qt-5/qopenglwidget.html#relation-to-qglwidget
> https://doc.qt.io/qt-5/qopenglwidget.html#differences-to-qglwidget
> https://doc.qt.io/qt-5/qopenglwidget.html#resource-initialization-and-cleanup

This seems to channel some of the issuses that have plagued the "fix"
"While the API is very similar, there is an important difference between the two: QOpenGLWidget always renders offscreen, using framebuffer objects. QGLWidget on the other hand uses a native window and surface. The latter causes issues when using it in complex user interfaces since, depending on the platform, such native child widgets may have various limitations, regarding stacking orders for example. QOpenGLWidget avoids this by not creating a separate native window."

**Comment by rwbaer** [comment]
I removed 2 lines added to glmodule.py by kshepherd2013   [glEnable(GL_DEPTH_TEST); ] that seemed to be responsible for the "shadow behavior" on windows.   I left his other modifications. 

 It will be important to find out if this breaks the Ubuntu 22.04 repair, or if they were unimportant to that repair.  It would be great if one of you could try Ubuntu 22.04 with the changes.

**Comment by joepal1976** [comment]
My workstation is still at ubuntu 20.04, I haven't had time to update it yet. And running a test in virtualbox isn't really fair.

With ubuntu 20.02, I don't get a shadow but the zoom bug is there

**Comment by rwbaer** [comment]
There is a small side effect of having removed these two lines in Windows. The clipping plane on the front of the face after a render is incorrect.  However this is only in the viewer tab after a render. The modeling tab and other tabs show the human as expected and seem to function normally with the exception of the zoom issue #208.  On Windows overall functionality is better with the lines gone because there is no seeing anything with the black shadow in front.

<img width="132" alt="image" src="https://user-images.githubusercontent.com/3155051/185748565-a5389c2a-bd91-4997-bee7-34ace6a21c9b.png">

**Comment by Aranuvir** [comment]
Same problem on my Linux-System, now. It looks like the shaded mesh is hidden behind the black object. Could it be a wrong buffer showing?

**Comment by rwbaer** [comment]
@Aranuvir - Just for clarity, do youn see the shadow with Ubuntu 22.04 or 20.04 (or both)?  Should we move the ksshepard2013  fix to a feature branch until we get this solved?  

Are we certain the old master code doesn't work on **_all_** Ubuntu 22.04 systems or is it just some of them? 

Either way, I think we do need to move forward on QOpenGLWidget (useful links in post [above ](https://github.com/makehumancommunity/makehuman/issues/206#issuecomment-1202850253)   It's a questionof what master is least toxic to sourcce users.

**Comment by rwbaer** [comment]
FWIW, with the current nightly I'm no longer getting the black shadow on Windows 11 when I render, but there is a new issue that mI describe in #225 along with a link to reports of back face culling issues by Ricardo2020 on the forum that may be related.  If you think they are the same issue, you can close this and continue there from the Windows point of view.
